src/app/lists/list/list-item-selection/list-item-selection.component.html

```diff
@@ -0,0 +1,25 @@
+<igx-input-group type="search" class="search">
+    <igx-prefix>
+        <igx-icon>search</igx-icon>
+    </igx-prefix>
+    <input #search igxInput placeholder="Search Contacts" [(ngModel)]="searchContact">
+    <igx-suffix *ngIf="search.value.length > 0" (click)="searchContact = null">
+        <igx-icon>clear</igx-icon>
+    </igx-suffix>
+</igx-input-group>
+
+<div class="list-sample">
+    <igx-list>
+        <igx-list-item isHeader="true">Contacts</igx-list-item>
+        <igx-list-item igxRipple
+                       igxRippleTarget=".igx-list__item-content"
+                       [ngClass]="contact.selected ? 'selected' : ''"
+                       (click)="selectItem(contact)"
+                       *ngFor="let contact of contacts | igxFilter: filterContacts;">
+            <igx-avatar igxListThumbnail [src]="contact.photo" roundShape="true"></igx-avatar>
+            <span igxListLineTitle>{{ contact.name }}</span>
+            <span igxListLineSubTitle>{{ contact.phone }}</span>
+            <igx-icon igxListAction [style.color]="contact.isFavorite ? 'orange' : 'lightgray'" (click)="toggleFavorite(contact)">star</igx-icon>
```

Comment:
clicking on the favorite start will also select the list item. IMO this is not ok. You can pass event to `toggleFavoriet` and call `event.stopPropagation` .